processing/src/main/java/org/apache/druid/frame/processor/ReturnOrAwait.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.frame.processor;
+
+import it.unimi.dsi.fastutil.ints.IntOpenHashSet;
+import it.unimi.dsi.fastutil.ints.IntSet;
+import it.unimi.dsi.fastutil.ints.IntSets;
+import org.apache.druid.java.util.common.IAE;
+import org.apache.druid.java.util.common.ISE;
+
+import javax.annotation.Nullable;
+import java.util.Objects;
+
+public class ReturnOrAwait<T>
```

Comment:
Explanation? On the surface, in an async system, one normally does one or the other. (Return what's available, or wait to fulfill the request.) Trying to sort out where this fits in...

Might be easier to just identify the problem this solves. Why, for example, do I want to wait for *all* channels rather than *any* channel?

Reply:
This class exists to be a return value from `FrameProcessor#runIncrementally`. I'll add a javadoc bit mentioning that. The idea is that this ReturnOrAwait instance tells the executor whether the processor wants to continue or not.

> Might be easier to just identify the problem this solves. Why, for example, do I want to wait for all channels rather than any channel?

Good question. Maybe we don't need the "wait for all channels" feature. Do you foresee any processors that might need to do such a thing? I don't have any examples offhand.

Reply:
I added a bunch of comments to this class that hopefully clear up the intended usage.